src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -236,28 +230,25 @@ def test_0004_k5login_user4_with_deleted_user3(multihost):
         for command in echo_cmds:
             cmd1 = multihost.client[0].run_command(command, raiseonerr=False)
 
-        user = 'foo3'
-        ssh_output = client.auth_from_client(user, 'Secret123')
+        ssh1 = check_login_client_bool(multihost, "foo3", "Secret123")
 
         ldap_cmd = f'ldapdelete -x -H ldap://{multihost.master[0].sys_hostname} \
                    -D "cn=Directory Manager" -w "Secret123" uid=foo3,ou=People,dc=example,dc=test'
-
         cmd2 = multihost.client[0].run_command(ldap_cmd, raiseonerr=False)
-
+        time.sleep(10)
         client_hostname = multihost.client[0].sys_hostname
-        ssh = pexpect_ssh(client_hostname, 'foo4', 'Secret123', debug=False)
-        ssh.login(login_timeout=10, sync_multiplier=1, auto_prompt_reset=False)
-        ssh.command("id > /tmp/accessProvider_id_krb5_004.out 2>&1 ")
-        ssh.command(f"ssh -o StrictHostKeyChecking=no -o PasswordAuthentication=no foo3@{client_hostname}\
-                    'id > /tmp/accessProvider_id_krb5_004.out 2>&1' ")
-        ssh.logout()
-
+        ssh = SSHClient(client_hostname, "foo4", "Secret123")
+        ssh.connect()
```

Comment:
What will happen if this fails? 
I do not see any error handling there so the truncate command will still not happen here if ssh fails.
Solution might be something along the lines:
```
try:
     ssh = SSHClient(client_hostname, "foo4", "Secret123")
     ssh.connect()
     ...
     ssh.close() 
except:
    pytest.fail("Error in connection via ssh as foo4")
finally:
    multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login", raiseonerr=False)
    ...
```